pkg/build/builder/docker.go

```diff
@@ -64,16 +60,10 @@ func (d *DockerBuilder) Build() error {
 		return err
 	}
 	glog.V(4).Infof("Starting Docker build from build config %s ...", d.build.Name)
-	var push bool
-
 	// if there is no output target, set one up so the docker build logic
-	// will still work, but we won't push it at the end.
+	// (which requires a tag) will still work, but we won't push it at the end.
 	if d.build.Spec.Output.To == nil || len(d.build.Spec.Output.To.Name) == 0 {
-		d.build.Spec.Output.To = &kapi.ObjectReference{
-			Kind: "DockerImage",
-			Name: noOutputDefaultTag,
-		}
-		push = false
+		d.build.Status.OutputDockerImageReference = d.build.Name
```

Comment:
i don't think the value actually matters in this case as long as it's a valid "docker tag" but i agree that setting it to build.name is probably not the right thing to do.  It probably should have remained as noOutputDefaultTag or some other clear constant.

I feel like there was some discussion around getting rid of the nodefaultoutput constant, but i can't remember the details.